src/main/java/duke/logic/UserInputParser.java

```diff
@@ -0,0 +1,212 @@
+
+import duke.CommonMethod;
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.HelpCommand;
+import duke.command.ListCommand;
+import duke.exception.InvalidInstructionFormatException;
+import duke.exception.InvalidInstructionLengthException;
+import duke.exception.MissingFieldException;
+import duke.exception.UnknownInstructionException;
+import duke.task.DeadlineTask;
+import duke.task.DukeTask;
+import duke.task.EventTask;
+import duke.task.TodoTask;
+
+
+/**
+ * Represents a Parser of the user Commands.
+ * It takes in the user's instructions and generates the corresponding command
+ * if it exists.
+ */
+public class UserInputParser {
+
+    // INSTRUCTIONS and related CONSTANTS
+    // Kept as Strings because of certain constants (indicators) that are related to certain instructions
+    // Convenient to use - final keeps them as constant values
+    private static final String BYE = "bye";
+    private static final String HELP = "help";
+    private static final String LIST = "list";
+    private static final String DONE = "done";
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String BY_INDICATOR = "/by";
+    private static final String EVENT = "event";
+    private static final String AT_INDICATOR = "/at";
+    private static final String DELETE = "delete";
+    private static final String FIND = "find";
+
+    /**
+     * Parses the user instruction into a <code>Command</code>.
+     * It breaks down the input <code>String</code> into an array and analyses the first word of it (tag).
+     * Referring to the tag, it performs an input validation to ensure the required <code>Strings</code>
+     * are present to generate the respective <code>Command</code>.
+     * NOTE THAT IT JUST CHECKS FOR THE PRESENCE OF THE STRING, AND NOT WHETHER THE STRING PRODUCES A VALID COMMAND.
+     * Finally, it returns a <code>Command</code> to the user.
+     *
+     * @param userInput <code>String</code> containing the user instruction.
+     * @return Command object denoting the corresponding command.
+     * @throws UnknownInstructionException       If instruction does not exist.
+     * @throws MissingFieldException             If the instruction has missing Strings.
+     * @throws InvalidInstructionLengthException If the instruction has correct format and fields, but incorrect input.
+     * @throws InvalidInstructionFormatException If the instruction has incorrect format.
+     */
+    public static Command parse(String userInput)
+            throws MissingFieldException, InvalidInstructionLengthException,
+            InvalidInstructionFormatException, UnknownInstructionException {
+
+        // Instruction Setup - split by whitespace to check
+        String instruction = userInput.trim();
```

Comment:
Perhaps you can consider another case - what happens if a user types in a command with the wrong capitalisation? Does your program still accept that as the correct command?